src/components/forms/VpnServerForm.js

```diff
@@ -0,0 +1,424 @@
+import { isIpStringInvalid, isRangeInvalid } from "../../lib/iam-utils";
+import {
+  buildFormDefaultInputMethods,
+  buildFormFunctions,
+} from "../component-utils";
+import { IcseSelect } from "../Dropdowns";
+import { IcseTextInput, IcseNameInput, IcseToggle } from "../Inputs";
+import { SecurityGroupMultiSelect } from "../MultiSelects";
+import { IcseFormGroup } from "../Utils";
+import IcseFormTemplate from "../IcseFormTemplate";
+import VpnServerRouteForm from "./VpnServerRouteForm";
+
+class VpnServerForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { ...this.props.data };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleNumberInputChange = this.handleNumberInputChange.bind(this);
+    this.handleMultiSelectChange = this.handleMultiSelectChange.bind(this);
+    this.handleToggle = this.handleToggle.bind(this);
+    this.handleAllowedIps = this.handleAllowedIps.bind(this);
+    buildFormFunctions(this);
+    buildFormDefaultInputMethods(this);
+  }
+
+  handleInputChange(event) {
+    let { name, value } = event.target;
+    let newState = { ...this.state };
+    //handle crn inputs
+    let crnList = value
+      ? value
+          .replace(/\s\s+/g, "") // replace extra spaces
+          .replace(/,(?=,)/g, "") // prevent null tags from
+          .replace(/[^\w,-:]/g, "")
+      : [];
+    if (name === "method") {
+      // Clear client_ca_crn when method changes
+      newState.method = value;
+      newState.client_ca_crn = "";
+    } else if (name === "vpc") {
+      // Clear subnet and security groups when vpc changes
+      newState.vpc = value;
+      newState.subnet = "";
+      newState.security_groups = [];
+    } else if (name === "certificate_crn") {
+      newState.certificate_crn = crnList;
+    } else if (name === "client_ca_crn") {
+      newState.client_ca_crn = crnList;
+    } else {
+      newState = { [name]: value };
+    }
+    this.setState(newState);
+  }
+
+  /**
+   * handle input change of number-only fields
+   * @param {event} event
+   */
+  handleNumberInputChange(event) {
+    let value = parseInt(event.target.value) || null;
+    if (value || event.target.value === "") {
+      this.setState({ [event.target.name]: value });
+    }
+  }
+
+  /**
+   * Handle input change for  client_dns_server_ips text field
+   * @param {event} event
+   */
+  handleAllowedIps(event) {
+    // removing white space and checking for empty value
+    let value = event.target.value.replace(/\s*/g, "");
+    if (value === "") value = null;
+    this.setState({ client_dns_server_ips: value });
+  }
+
+  handleMultiSelectChange(name, value) {
+    this.setState(this.setNameToValue(name, value));
+  }
+
+  handleToggle(name) {
+    this.setState(this.toggleStateBoolean(name, this.state));
+  }
+
+  render() {
+    let composedId = `vpn-server-form-${this.props.data.name}`;
+    let classNameModalCheck = this.props.isModal
```

Comment:
i think we can just use `fieldWidthSmaller` for all the normal fields, load balancer form uses only `fieldWidthSmaller`